components/ai_chat/core/browser/conversation_handler_unittest.cc

```diff
@@ -1304,6 +1297,36 @@ TEST_F(ConversationHandlerUnitTest_NoAssociatedContent, GenerateQuestions) {
   testing::Mock::VerifyAndClearExpectations(engine);
 }
 
+TEST_F(ConversationHandlerUnitTest_NoAssociatedContent,
+       GeneratesQuestionsByDefault) {
+  EXPECT_EQ(4u, conversation_handler_->GetSuggestedQuestionsForTest().size());
+}
+
+TEST_F(ConversationHandlerUnitTest_NoAssociatedContent,
+       SelectingDefaultQuestionSendsPrompt) {
+  conversation_handler_->SetSuggestedQuestionForTest("the thing",
+                                                     "do the thing!");
+  auto suggestions = conversation_handler_->GetSuggestedQuestionsForTest();
+  EXPECT_EQ(1u, suggestions.size());
+
+  // Mock engine response
+  MockEngineConsumer* engine = static_cast<MockEngineConsumer*>(
+      conversation_handler_->GetEngineForTesting());
+
+  // The prompt should be submitted to the engine, not the title.
+  EXPECT_CALL(*engine,
+              GenerateAssistantResponse(false, StrEq(""), _, "do the thing!",
+                                        StrEq(""), _, _));
+
+  testing::Sequence s;
+  conversation_handler_->SubmitHumanConversationEntry("the thing");
+  task_environment_.RunUntilIdle();
```

Comment:
In this case, we should be able to use RunLoop and have it Quit inside the EXPECT_CALL for GenerateAssistantResponse that call marks the end of when we care? I think we are meant to reduce our usage of RunUntilIdle unless it's neccessary...